src/NServiceBus.Metrics/CriticalTime/CriticalTimeMetricBuilder.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Threading.Tasks;
+using Metrics;
+using NServiceBus;
+using NServiceBus.Features;
+
+class CriticalTimeMetricBuilder : IMetricBuilder
+{
+    public void WireUp(FeatureConfigurationContext featureConfigurationContext, MetricsContext metricsContext, Unit messagesUnit)
+    {
+        var criticalTimeTimer = metricsContext.Timer("Critical Time", messagesUnit);
```

Comment:
```
using(timer.NewContext("Blah")
{
  // ...
}
```

results in a call to `timer.Record(end - start, TimeUnit, "Blah");`

we're just shortcutting it because we've already captured start and end. 

[This is the code](https://github.com/Recognos/Metrics.NET/blob/1533c9a78a23492b3c7de08e41720b6cafb10f72/Src/Metrics/Timer.cs#L136) where the timer context is disposed and the call to `Record` is made. 

We could do it as a pipeline behavior in order to be able to use the `TimerContext`.